### PR TITLE
[MID-32] add pre-push and commit-msg hooks to check branch and commit name

### PR DESCRIPTION
**Pull request by IvanKoshan**
this is a pr for task: https://trello.com/c/jADOe5NX/32

### CODE REVIEW

**Review by alex-popov-tech — APPROVED** [review]

File: .husky/pre-push.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const branchName = process.argv[2];
+
+const testRegExp = /^MID-[0-9]+_([a-zA-Z0-9]+-?)+[^-]$/g;
```
Comment (Mykhailo-Pasternak) [inline .husky/pre-push.js:3]: 🟢 Checked writing regexp through GPT.  👍

File: .husky/commit-msg.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+const fs = require('fs');
+
+const commitMsgFile = process.argv[2];
+const commitMsg = fs.readFileSync(commitMsgFile, 'utf8');
+const testRegExp = /^\[MID-[0-9]+\] /g;
```
Comment (Mykhailo-Pasternak) [inline .husky/commit-msg.js:5]: 🟢 Checked

**Review by Mykhailo-Pasternak — APPROVED** [review]

**Review by dmitrenko-v — APPROVED** [review]

File: .husky/commit-msg.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+const result = testRegExp.exec(commitMsg);
+if (result === null) {
+  console.log('Commit message should be of the form: "[MID-<ticket number>] <commit message text>"');
+  console.log('Example: "[MID-123] add this cool feature"');
+  console.log(`Provided commit message ${commitMsg}`);
```
Comment (yataras) [inline .husky/commit-msg.js]: Consider a minor change: `
```suggestion
  console.log(`Provided: ${commitMsg}`);
```

File: .husky/commit-msg.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  process.exit(1);
+}
+process.exit(0);
```
Comment (yataras) [inline .husky/commit-msg.js:14]: Consider including additional check - for max or min words/character in commit message. It should make commits smaller, smarter and consistent.

File: .husky/pre-push.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+const result = testRegExp.exec(branchName);
+if (result === null) {
+  console.log('Branch name should be of the form: "MID-<ticket number>_<branch-name-text-in-kebab-case>"');
+  console.log('Example: "MID-123_add-this-cool-feature"');
+  console.log(`Provided branch name ${branchName}`);
```
Comment (yataras) [inline .husky/pre-push.js]: Consider similar suggestions like for commit messages

File: package.json
Comment (yataras) [inline package.json:1]: @IvanKoshan What about suggested in the ticket packages? ([commitlint](https://www.npmjs.com/package/commitlint) and [npm: branch-name-lint](https://www.npmjs.com/package/branch-name-lint)).
Could you please share if there are plans to implement these packages in our workflow in the future? Or have these suggestions been set aside for the time being?

**Review by yataras — CHANGES_REQUESTED** [review]

File: .husky/commit-msg.js (hunk 4 above)
Comment (IvanKoshan) [reply .husky/commit-msg.js:14]: I've asked our Tech Lead if we need this change. Currently we decided that it's unnecessary to add such restriction.

File: .husky/pre-push.js (hunk 5 above)
Comment (IvanKoshan) [reply .husky/pre-push.js]: same here

File: package.json
Comment (IvanKoshan) [reply package.json:1]: At the moment, we'll stick to the current conventions

File: .husky/commit-msg.js (hunk 3 above)
Comment (IvanKoshan) [reply .husky/commit-msg.js]: Changed it

**Review by yataras — APPROVED** [review]

**Pull request merged**